packages/utils/addon/decorator.js

```diff
@@ -61,11 +61,16 @@ function convertStage1ToStage2(desc) {
     let kind = kindForDesc(desc);
     let placement = placementForKind(kind);
 
+    let { initializer } = descriptor;
+    delete descriptor.initializer;
```

Comment:
I don't think we _need_ this `delete`, and my understanding is that generally it's best to avoid them when possible. Messes with the shape of objects and can cause deopts. I think it would be fine to pull the initializer off and set it back later.

Reply:
Correct. We don't need it. I've removed it, so that users don't accidentally use or try to override this initializer in the wrong place.

I'll remove the `delete`.